plio/signals.py

```diff
@@ -31,8 +31,12 @@ def question_update_cache(sender, instance, **kwargs):
 
 
 @receiver([post_save], sender=Question)
-def question_delete_image_on_deletion(sender, instance, **kwargs):
-    # check if any image is linked to the instance
+def delete_linked_image_on_question_deletion(sender, instance, **kwargs):
+    # since we are using soft deletion, the `post_delete` signal is never called
+    # also, the way deletion works under the hood when using soft delete is that
+    # just sets the `deleted` attribute and updates the instance. so, by default,
+    # the deleted attribute is None. When we execute soft_delete, that attribute gets set
```

Comment:
Some inconsistency
`so, be default...` here, a new sentence starts with a lower case letter
`When we execute ... ` here, a new sentence starts with an upper case letter

We should decide on a format or we should ignore such things and not include them in the review.

Mentioning this because in another of backend PR of mine, the review included comments starting with caps and that they should be turned to lower case

I don't mind too much about the upper/lower casing of comments so let me know whatever you decide